### PR TITLE
fix: use bedrock cost func for concurrent activation

### PR DESCRIPTION
**Pull request by danyalprout**
### Description
Address Sebs comment [here](https://github.com/ethereum-optimism/op-geth/pull/249#discussion_r1557419276).

### CODE REVIEW

File: core/types/rollup_cost.go [hunk 1]
```diff
@@ -139,21 +155,12 @@ func NewL1CostFunc(config *params.ChainConfig, statedb StateGetter) L1CostFunc {
 				l1CostFastlzCoef,
 				l1CostTxSizeCoef,
 			)
+		} else if config.IsOptimismEcotone(blockTime) {
+			l1BaseFeeScalar, l1BlobBaseFeeScalar := extractEcotoneFeeParams(l1FeeScalars)
+			return newL1CostFuncEcotone(l1BaseFee, l1BlobBaseFee, l1BaseFeeScalar, l1BlobBaseFeeScalar)
+		} else {
+			return newL1CostFuncBedrock(config, statedb, blockTime)
 		}
```
Comment (roberto-bayardo) [inline core/types/rollup_cost.go]:
```suggestion
		return newL1CostFuncBedrock(config, statedb, blockTime)
```

File: core/types/rollup_cost.go [hunk 2]
```diff
@@ -128,6 +128,22 @@ func NewL1CostFunc(config *params.ChainConfig, statedb StateGetter) L1CostFunc {
 		l1FeeScalars := statedb.GetState(L1BlockAddr, L1FeeScalarsSlot).Bytes()
 		l1BlobBaseFee := statedb.GetState(L1BlockAddr, L1BlobBaseFeeSlot).Big()
 		l1BaseFee := statedb.GetState(L1BlockAddr, L1BaseFeeSlot).Big()
+
+		if config.IsOptimismEcotone(blockTime) {
+			// Edge case: the very first Ecotone block requires we use the Bedrock cost
+			// function. We detect this scenario by checking if the Ecotone parameters are
+			// unset. Note here we rely on assumption that the scalar parameters are adjacent
+			// in the buffer and l1BaseFeeScalar comes first.
+			// We need to check this prior to other forks, as the first block of Fjord and Ecotone could be the  same
```
Comment (roberto-bayardo) [inline core/types/rollup_cost.go]: super nit pick: could you reformat the entire comment with the same line width?  (L137 is substantially longer than the others here.)

**Comment by roberto-bayardo** [comment]
This looks good.. did you confirm the monorepo change is handling this identically?

**Comment by danyalprout** [comment]
It should be identical, except for when the first block is concurrent. I'm planning to add tests in the monorepo for this case

**Review by roberto-bayardo — APPROVED** [review]

**Pull request merged**